test/testdata/infer/is_a_variable.rb

```diff
@@ -0,0 +1,34 @@
+# typed: true
+extend T::Sig
+
+# At time of writing, this test captures buggy behavior, and is just meant to
+# alert the author if their changes affect this behavior.
+# https://github.com/sorbet/sorbet/issues/4358
```

Comment:
Heads up @elliottt I added some tests to capture the existing behavior and filed a ticket.